Description: Use GNU dd where uutils dd breaks
 Workaround test failure caused by uutils dd
Author: Nick Rosbrook <enr0n@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/openssh/+bug/2125943
Forwarded: no
Last-Update: 2025-09-29
--- a/regress/putty-transfer.sh
+++ b/regress/putty-transfer.sh
@@ -13,7 +13,14 @@
 	comp="0 1"
 fi
 
-for c in $comp; do 
+# Workaround LP: #2125943
+if [ -f "/usr/bin/gnudd" ]; then
+    DD="/usr/bin/gnudd"
+else
+    DD="/usr/bin/dd"
+fi
+
+for c in $comp; do
 	verbose "$tid: compression $c"
 	rm -f ${COPY}
 	cp ${OBJ}/.putty/sessions/localhost_proxy \
@@ -29,7 +36,7 @@
 	for s in 10 100 1k 32k 64k 128k 256k; do
 		trace "compression $c dd-size ${s}"
 		rm -f ${COPY}
-		dd if=$DATA obs=${s} 2> /dev/null | \
+		${DD} if=$DATA obs=${s} 2> /dev/null | \
 			env HOME=$PWD ${PLINK} -load compression_$c \
 			    -batch -i ${OBJ}/putty.rsa2 \
 			    "cat > ${COPY}"
--- a/regress/transfer.sh
+++ b/regress/transfer.sh
@@ -10,10 +10,17 @@
 fi
 cmp ${DATA} ${COPY}		|| fail "corrupted copy"
 
+# Workaround LP: #2125943
+if [ -f "/usr/bin/gnudd" ]; then
+    DD="/usr/bin/gnudd"
+else
+    DD="/usr/bin/dd"
+fi
+
 for s in 10 100 1k 32k 64k 128k 256k; do
 	trace "dd-size ${s}"
 	rm -f ${COPY}
-	dd if=$DATA obs=${s} 2> /dev/null | \
+	${DD} if=$DATA obs=${s} 2> /dev/null | \
 		${SSH} -q -F $OBJ/ssh_proxy somehost "cat > ${COPY}"
 	if [ $? -ne 0 ]; then
 		fail "ssh cat $DATA failed"
